js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js

```diff
@@ -1,504 +1,530 @@
-			} else {
-				if (self.address && self.address.trim().length > 3) {
-					mapServiceComposite.suggestAddresses(self.address, function(autoCompleteResults) {
-						self.autocompleteResults = autoCompleteResults;
-						$scope.$apply();
-					});
-				}
-			}
-		}
-
-		function autocompletePlaceChangedHandler() {	
-			var place = autocomplete.getPlace();
-			if (place.formatted_address)
-				geocodeAndMarkAddress(place.formatted_address);			
-		}
-
-		function autocompleteResultButtonKeyboardNavigationHandler(event) {
-			var keycode = event.which || event.keyCode;
-			var $target = angular.element(event.target);
-
-			if (!$target.is('.autocomplete-results button')) {
-				return;
-			}
-			
-			if (keycode === 27 && angular.element('.autocomplete-results').is(':visible')) {
-				self.autocompleteResults = [];
-				$scope.$apply();
-			}
-
-			event.preventDefault();			
-
-			if (keycode === 13 || keycode === 32) {
-				$target.trigger('click');
-			}
-
-			if (keycode === 40) {
-				$target.parent().next().find('button').focus();
-			}
-
-			if (keycode === 38) {
-				$target.parent().prev().find('button').focus();
-			}
-		}
-
-		function animalTypeSuccessHandler(response) {
-			self.animalTypeData = response.data;
-		}
-
-		function breedSuccessHandler(response) {
-			self.animalBreedData = response.data;
-		}
-
-		function categorySuccessHandler(response) {
-			self.categoryData = response.data;
-			if (categoryId)
-				autoSelectCategories(categoryId);
-		}
-
-		function colorSuccessHandler(response) {
-			self.animalColorData = response.data;
-		}
-
-		function displayAddressError() {
-			angular.element('#map').closest('cs-form-control').addClass('error');
-			$scope.citySourcedReporterForm.address.$setDirty();
-			self.address = '';
-			$scope.$apply();
-		}
-
-		function errorHandler(err) {
-			console.log(err);
-		}
-
-		function mapClickHandler(event) {			
-			var $wrapper = angular.element('#map').closest('cs-form-control'),
-				addressField = $scope.citySourcedReporterForm.address;
-			
-			self.autocompleteResults = [];
-			self.latitude = event.latLng.lat();
-			self.longitude = event.latLng.lng();
-
-			mapServiceComposite.reverseGeocode(self.latitude, self.longitude, function (response) {
-				$wrapper.removeClass('error');
-				mapServiceComposite.createMarker(map, self.latitude, self.longitude);
-				self.address = response.address.Street.toLowerCase() + ', ' + response.address.City.toLowerCase() + ', ' + response.address.State.toUpperCase();
-				$scope.$apply();
-			}, function(err) {
-				$wrapper.addClass('error');
-				addressField.$setDirty();
-				self.address = '';				
-				$scope.$apply();
-			});
-		}
-
-		function petTypeSuccessHandler(response) {
-			self.petTypeData = response.data;
-		}
-
-		function preventSubmitOnEnterPressHandler(event) {
-			var keyCode = event.which || event.keyCode;
-
-			if (keyCode === 13) {
-				event.preventDefault();
-			}		
-		}
-
-	}
+  'use strict';
+
+  function Data(arr) {
```

Comment:
Why are the Data function and the provider here? You have them elsewhere, so just inject it from the other file.